docs/blog/rendering-images/index.qmd

```diff
@@ -0,0 +1,292 @@
+construct invalid paths. A good practice to mitigate this is to use Python's built-in
+[Pathlib](https://docs.python.org/3/library/pathlib.html) module to construct paths first and then
+convert them to strings.
+:::
+
+Now that the `case1` column of `metro_mini1` contains the correct local file paths, we can use
+`GT.fmt_image()` to render the images by providing `"case1"` as the first argument:
+```{python}
+GT(metro_mini1).fmt_image("case1")
+```
+
+### Case 2: Full HTTP/HTTPS URLs
+**Case 2** demonstrates how to simulate a column containing strings representing HTTP/HTTPS URLs. We'll
+use the same images as in **Case 1**, but this time, retrieve them from the Great Tables GitHub repository:
+```{python}
+img_url_paths = "https://raw.githubusercontent.com/posit-dev/great-tables/refs/heads/main/great_tables/data/metro_images"
+```
+
+Preparing a Python dictionary as input for creating a `Pandas` DataFrame follows a similar process
+to **Case 1**. However, in this case, the new key is named `case2`:
+```{python}
+# | code-fold: true
+metro_mini2 = pd.DataFrame(
+    {
+        **data,
+        "case2": [
+            ", ".join(f"{img_url_paths}/metro_{item}.svg" for item in row.split(", "))
+            for row in data["lines"]
+        ],
+    }
+)
+metro_mini2
+```
+
+In the same way, we can use `GT.fmt_image()` to render the images by passing `"case2"` as the first
+argument:
+```{python}
+GT(metro_mini2).fmt_image("case2")
+```
+
+
+### Case 3: File Names with the `path=` Argument
+**Case 3** demonstrates how to simulate a column containing file names. The preparation for creating
+a `Pandas` DataFrame is similar to the previous cases but simpler. This time, we only need to construct
+file names. The new key in the dictionary is named `"case3"`:
+```{python}
+# | code-fold: true
+metro_mini3 = pd.DataFrame(
+    {
+        **data,
+        "case3": [
+            ", ".join(f"metro_{item}.svg" for item in row.split(", ")) for row in data["lines"]
+        ],
+    }
+)
+metro_mini3
+```
+
+Now we can use `GT.fmt_image()` to render the images by passing `"case3"` as the first argument and
+specifying either `img_local_paths` or `img_url_paths` as the `path=` argument:
+```{python}
+GT(metro_mini3).fmt_image("case3", path=img_local_paths)  # equivalent to `Case 1`
+
+GT(metro_mini3).fmt_image("case3", path=img_url_paths)    # equivalent to `Case 2`
+```
+
+After exploring **Case 1** and **Case 2**, you’ll likely appreciate the functionality of the `path=`
+argument. However, manually constructing file names can still be a bit tedious. If your file names
+follow a consistent pattern, the `file_pattern=` argument can simplify the process. Let’s see how
+this works in **Case 4** below.
+
+### Case 4: File Names Using Both the `path=` and `file_pattern=` Arguments
+**Case 4** demonstrates how to simulate a column containing only the variable part of a string pattern.
```

Comment:
Similar to case 3, the why of case 4 up top might help people see what they'll get out of this section. something like..?:

> **Case 4** demonstrates how to use `path=` and `file_pattern=` to specify images whose names are inside a common name. For example, you might use `file_pattern="metro_{}.svg"`, to specify `"metro_1.svg"`, `"metro_2.svg"`, etc..